src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -42,6 +44,25 @@ export const wdsToEntityServiceMetadata = (wdsSchema: RecordTypeSchema[]): Entit
   }, keyedSchema)
 }
 
+export const relationUriScheme = 'terra-wds'
+
+// Callers outside this module should not call this function. It returns a string array of size 2
+// iff it looks like a valid relation URI; else it returns an empty array.
+// Returning the array prevents callers from string-parsing the URI multiple times.
+// This function does not verify that the record type and record id are syntactically
+// valid according to WDS - for instance, do they contain special characters?
+// WDS should own that logic. Here, we only check if the type and id are nonempty.
+const getRelationParts = (val: unknown): string[] => {
+  if (_.isString(val) && val.startsWith(`${relationUriScheme}:/`)) {
+    const parts: string[] = val.substring(relationUriScheme.length + 2).split('/')
+    if (parts.length === 2 && _.every(part => !!part, parts)) {
+      return parts
+    }
+    return []
```

Comment:
Just to fully understand, would this be a situation where an end user wanted some relational value, but does not use the `terra-wds:/` pattern?

Reply:
not quite. The ugliness of this `getRelationParts` is a result of in-browser performance optimization. This gets called for every attribute value, so I want to make it as performant as possible.

Checking to see if an attribute value is a valid relation or not requires a bunch of string manipulation: does it start with `terra-wds`? If so, does it then have two non-empty path parts separated by a slash?

I originally tried writing prettier code which did:
1. test to see if the value is a relation; return true or false
2. if so, translate it into an `EntityReference` object

But, both 1 and 2 above needed to perform the same string manipulation, which I really wanted to avoid. So I modified it so that:
1. test to see if the value is a relation; return either a populated array of the URI parts (representing true) or an empty array (representing false)
2. if truthy, translate the populated array into an `EntityReference` object

Thus, `getRelationParts` is kinda wonky and returns an empty array at times.

Reply:
makes sense -- thanks for the explanation!